File: PathfinderPlugin/src/main/java/org/bukkit/craftbukkit/v1_12_R1/pathfinding/CraftNavigation.java
package org.bukkit.craftbukkit.v1_12_R1.pathfinding;

import net.minecraft.server.v1_12_R1.NavigationAbstract;
import org.bukkit.Location;
import org.bukkit.craftbukkit.v1_12_R1.entity.CraftEntity;
import org.bukkit.entity.Entity;
import org.bukkit.pathfinding.Navigation;

/**
 * Created by ysl3000
 */
public class CraftNavigation implements Navigation {

    private NavigationAbstract navigationAbstract;

    public CraftNavigation(NavigationAbstract navigationAbstract) {
        this.navigationAbstract = navigationAbstract;
    }

    @Override
    public void moveTo(Location location) {
        navigationAbstract.a(location.getX(), location.getY(), location.getZ());
    }

    @Override
    public void moveTo(Entity entity) {
        navigationAbstract.a(((CraftEntity) entity).getHandle());
    }

    @Override
    public boolean moveTo(Location location, double speed) {
        return navigationAbstract.a(location.getX(), location.getY(), location.getZ(), speed);
    }

    @Override
    public boolean moveTo(Entity entity, double speed) {
        return navigationAbstract.a(((CraftEntity) entity).getHandle(), speed);
    }

    @Override
    public void setSpeed(double speed) {
        navigationAbstract.a(speed);
    }

    @Override
    public boolean isMovementSet() {
        return navigationAbstract.o();
    }

    /**
     * sets active PathEntity to null
     */
    @Override
    public void clearPathEntity() {
        navigationAbstract.p();
    }





    /**
     * The maximal distance the pathfinder search
     *
     * @return distance
     */
    @Override
    public float getPathSearchRange() {
        return navigationAbstract.i();
    }

}

File: Pathfinder_1_13_1/src/main/java/com/github/ysl3000/bukkit/pathfinding/craftbukkit/v1_13_R2/pathfinding/CraftNavigation.kt
package com.github.ysl3000.bukkit.pathfinding.craftbukkit.v1_13_R2.pathfinding

import com.github.ysl3000.bukkit.pathfinding.AbstractNavigation
import org.bukkit.Location
import org.bukkit.craftbukkit.v1_13_R2.entity.CraftEntity
import org.bukkit.entity.Entity

import com.github.ysl3000.bukkit.pathfinding.pathfinding.Navigation

import net.minecraft.server.v1_13_R2.NavigationAbstract

class CraftNavigation(private val navigationAbstract: NavigationAbstract) : AbstractNavigation(
        doneNavigating={navigationAbstract.p()},
        pathSearchRange={navigationAbstract.j()},
        moveToPositionU={x, y, z -> navigationAbstract.a(x,y,z) },
        moveToPositionB={x, y, z, speed -> navigationAbstract.a(x,y,z,speed) },
        moveToEntityU={entity -> navigationAbstract.a((entity as CraftEntity).handle) },
        moveToentityB={entity, speed -> navigationAbstract.a((entity as CraftEntity).handle,speed) },
        speedU={speed -> navigationAbstract.a(speed) },
        clearPathEntityU={navigationAbstract.q()}
)
